### ISSUE TITLE
Add more specific business types?

### ISSUE THREAD

**Issue by robneu** [NateWr/business-profile#31]
Someone recently asked me if it's possible to add a more specific business type such as [InsuranceAgency](https://schema.org/InsuranceAgency). A method for doing it via a filter didn't jump out at me and I was wondering if you'd be willing to add these sub-types in to the default options array.

If so, I'll gather up a list of all of them and set up a PR. I can also add some filters to make it easier for developers to add new sub-types if they're added to the spec before we have a chance to include them.

**Comment by robneu** [comment]
Actually, on 2nd glance it looks like [this filter](https://github.com/NateWr/business-profile/blob/5acc86e072cb99e9ad671974532515f862629698/includes/Settings.class.php#L309) would work for extending the settings page?

Either way, I can add in the extra business types to the core plugin if you'd like.

**Comment by NateWr** [comment]
The entire settings interface is filtered, which allows you to dip in and change the options available for any of the items. [This gist](https://gist.github.com/NateWr/17402282adce7c0284cf) demonstrates how you'd do it.

There's a basic function for pulling [specific settings out](https://github.com/NateWr/business-profile/blob/master/includes/Settings.class.php#L46-L65), that just allows for better backwards compatibility. But in Restaurant Reservations I have [added a filter](https://github.com/NateWr/restaurant-reservations/blob/master/includes/Settings.class.php#L190-L209) which allows you to globally override a setting anywhere. Maybe this would be a good idea for Business Profile if you wanted to ensure the value would always be the same regardless of what the user puts in.

**Comment by NateWr** [comment]
I'm actually keen on keeping the list short, just because I think the options available via Schema are kind of overwhelming for the regular user and therefore susceptible to mis-use. If you think there's a good way to make an extended list accessible without presenting hundreds of choices to every user, I'd definitely be open to it (maybe a pop-up list or something?).

**Comment by robneu** [comment]
Hmm... Isn't Select2 in core now? I feel like it was added at some point for some reason....

If so, we could use that to make the list less terrible to search through. Alternatively, since it can be filtered we could just make a small add-on plugin to enable business sub-types or something.

**Comment by NateWr** [comment]
No, Select2 was found to be not accessible enough. It's a tough problem. I'm just sitting down for dinner but keep this on my radar and I'll keep thinking about it.

**Comment by robneu** [comment]
Sounds good! For now I'll just filter it for this particular use case. Just drop a note on here if you'd like me to set up a PR at some point.

**Comment by robneu** [comment]
Just a thought:

What about a conditional sub-select. Basically you select an option from the main list and if that option has sub-types a secondary select would appear. Should be doable with some JS show/hide hackery of some sort...

**Comment by NateWr** [comment]
Yeah that could work. It would take a bit to get the accessibility right -- ensuring the right visibility and `aria-live` attributes -- which I'm still a bit vague on. I'm tempted to just provide a single link that pops out a long list you could select from, as I find it easy to make modals or side-panels accessible. But the idea of sub-types would probably ease a lot of the pain of selecting from a long list.

**Comment by NateWr** [comment]
Added a filter to the available schema types for now: https://github.com/NateWr/business-profile/commit/f41e4c56f5c05969077e8a994c69c4c63516a45c
